src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php

```diff
@@ -114,6 +142,8 @@ protected function doGetArguments(Request $request, $controller, array $paramete
                 $arguments[] = $request;
             } elseif ($param->isDefaultValueAvailable()) {
                 $arguments[] = $param->getDefaultValue();
+            } elseif ($this->supportsParameterType && null !== ($type = $param->getType()) && $type->allowsNull()) {
```

Comment:
[That's because `getType` is only available in 7.0+](http://php.net/manual/en/reflectionparameter.gettype.php)